### PR TITLE
Add taint parameter and change rules

### PR DESCRIPTION
**Pull request by ErebusZ**
The change in the workflow is due to a breaking change in the runner images.
see [issue](https://github.com/actions/runner-images/issues/8952).

### CODE REVIEW

**Review by mohsinenar — APPROVED** [review]

File: app/src/main/java/co/ostorlab/insecure_app/MainActivity.java [hunk 1]
```diff
@@ -62,8 +66,10 @@ public void onClick(View view) {
         runAllButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
+                String user_input = inputField.getText().toString();
                 outputView.setText("Running \n");
-                executeAllRules();
+
+                executeAllRules(user_input);
```
Comment (3asm) [inline app/src/main/java/co/ostorlab/insecure_app/MainActivity.java:73]: Should come from an intent to make it dangerous.

File: app/src/main/java/co/ostorlab/insecure_app/bugs/calls/WebviewInsecureSettings.java [hunk 2]
```diff
@@ -16,14 +16,20 @@ public String getDescription() {
...
+    public void run(String user_input) throws Exception {
         ContextCompat.getMainExecutor(getContext()).execute(()  -> {
             WebView webView  = new WebView(getContext());
             webView.getSettings().setJavaScriptEnabled(true);
             webView.getSettings().setAllowUniversalAccessFromFileURLs(true);
             webView.getSettings().setAllowFileAccess(true);
             webView.setWebContentsDebuggingEnabled(true);
-            webView.loadUrl("http://www.ostorlab.co");
-                });
+            if (user_input.length() != 0){
```
Comment (3asm) [inline app/src/main/java/co/ostorlab/insecure_app/bugs/calls/WebviewInsecureSettings.java]: isEmpty instead of length() != 0.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: app/src/main/java/co/ostorlab/insecure_app/MainActivity.java [hunk 3]
```diff
@@ -39,41 +40,56 @@
...
+        if (intent.hasExtra("user_input")) {
+            user_input = intent.getStringExtra("user_input");
+            executeAllRules(user_input);
+            triggerFlutter(user_input);
+        }
+        else {
+            user_input = "";
+            // Trigger flutter directly when the app starts.
+            triggerFlutter(user_input);
+        }
```
Comment (3asm) [inline app/src/main/java/co/ostorlab/insecure_app/MainActivity.java]: And the rules. Why are the rules not executed here as well with default input.

File: app/src/main/java/co/ostorlab/insecure_app/MainActivity.java [hunk 4]
```diff
@@ -39,41 +40,56 @@
...
+
+        final Intent intent = getIntent();
+        String user_input;
+        if (intent.hasExtra("user_input")) {
+            user_input = intent.getStringExtra("user_input");
+            executeAllRules(user_input);
+            triggerFlutter(user_input);
+        }
+        else {
+            user_input = "";
```
Comment (3asm) [inline app/src/main/java/co/ostorlab/insecure_app/MainActivity.java]: Inline user_input in this case.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by amine3 — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**